### PR TITLE
Add UI to handle proposed sessions 

### PR DESCRIPTION
**Pull request by tvararu**
This allows nurses to transfer patients from one session to another after they upload a class list.

TODO:

- [x] Handle scenario where session they're coming from has vaccination records (create new patient_session)
- [x] Count of moving out/in
- [x] End to end test

### Screenshots

Screenshot 2024-10-23 at 10 46 25
Screenshot 2024-10-23 at 10 46 38

Screenshot 2024-10-23 at 10 46 41

### CODE REVIEW

File: app/models/patient_session.rb [hunk 1]
```diff
@@ -126,4 +128,23 @@ def latest_triage
...
   end
+
+  def pending_transfer?
+    proposed_session_id.present?
+  end
+
+  def confirm_transfer!
+    return unless pending_transfer?
+
+    if vaccination_records.any?
```
Comment (thomasleese) [inline app/models/patient_session.rb:139]: We also need to consider gillick assessments here, we should use `!safe_to_destroy?` I think.

File: app/models/patient_session.rb [hunk 2]
```diff
@@ -126,4 +128,23 @@ def latest_triage
...
+      PatientSession.create!(patient:, session: proposed_session)
+      update!(proposed_session: nil)
+    else
+      update!(session: proposed_session, proposed_session: nil)
```
Comment (thomasleese) [inline app/models/patient_session.rb:143]: I'm wondering whether it's better to always create a new one and delete old ones? It means `created_at` can be relied on to mean when the patient was added to the session.

Comment (tvararu) [reply app/models/patient_session.rb:143]: Hmm, yeah that's a good idea

Comment (tvararu) [reply app/models/patient_session.rb]: Updated!

**Review by thomasleese — APPROVED** [review]

**Pull request merged**